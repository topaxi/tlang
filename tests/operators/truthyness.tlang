fn map([], _) { [] }
fn map([x, ...xs], f) { [f(x), ...map(xs, f)] }

fn to_bool(v) { !!v }
fn void() {}

let truthy_values = [true, 1, [1], { a: 1 }] |> map(fn(v) { [v, to_bool(v)] });
let falsy_values = [void(), false, 0, [], {}] |> map(fn(v) { [v, to_bool(v)] });

log("Truthy values");
log("============");
log(truthy_values);

log("");

log("Falsy values");
log("============");
log(falsy_values);
