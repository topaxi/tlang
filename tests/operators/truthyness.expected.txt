Truthy values
============
[[true, true], [1, true], [[1], true], [Dict { a: 1 }, true]]

Falsy values
============
[[nil, false], [false, false], [0, false], [[], false], [Dict { }, false]]
